### DISCUSSION TITLE
[Announcement] A new Web app is now available

### DISCUSSION THREAD

**Discussion by yume-chan** [yume-chan/ya-webadb#576] [Announcement]
## Summary

A new, user friendly, Web app for interacting Android devices is under development.

Follow the progress at https://tangoapp.dev/

## Background

As I have said many times before, the demo (https://tango-adb.github.io/old-demo/, source code at https://github.com/tango-adb/old-demo) is only a demo. It's used in my development to test new features, and helps developers to learn how to integrate Tango library into their own projects.

However, more and more features were added over the time per user requests. I can still use it in development, but other people are having difficulties understanding such a complex project. On the other hand, the user experience isn't satisfying neither.

As a result, I decided to create a new Web app for Tango, hiding all the technical details, and exposing more features the users need, even if they don't fit Tango's original project goals.

## (Not) Open Source

Another important thing about the new app is that it will be close-sourced. Mainly due to three reasons:

1. This repository hasn't received any code contributions on new features (there are several PRs fixing small bugs, and I do appreciate them)
2. Someone has shamelessly copied the demo and put ADs in it. The credits were only added after people on XDA Developers noticed my original repository. (This project is under MIT license, so you can use all the source code freely, even commercially. Just remember to add the [LICENSE](https://github.com/yume-chan/ya-webadb/blob/main/LICENSE) text somewhere in your app, and if you want, post here to share with everyone).
3. The new app has features that are out of this project's scope. For example, there are some very badly written Kotlin code (I don't know Kotlin) to retrieve some information. I want to keep this repository focused on what Google ADB can do, using only TypeScript, and keep an overall high code quality.

## Commercialize

Although the new app will be close-sourced, there is **no** plan to commercialize it in near future. I don't want to put ADs in it to ruin the beautiful design, nor charge for features that are easily replicable. The app is freely hosted on Vercel, the only cost is the domain name (about 10 USD per year) and my time. If you want, you can donate (https://opencollective.com/ya-webadb) to support this project.

## Features

* Most features from the "File Manager" and "Scrcpy" pages of the old demo will be re-implemented in the new app.
* An app manager will be added (previously considered out-of-scope for this project because it requires calling Android API using Java/Kotlin code).
* Feature requests in the issue tracker that are still open has a higher priority to be added in future.

To name some:

- Device Info
   - [x] Device name (Bluetooth name) 
   - [x] Screenshot
   - [x] Connected WiFi network name and IP address
   - [x] Battery level
   - [x] Battery charging state
   - [x] Storage usage
   - [x] RAM usage
   - [ ] Control media playback
   - [ ] Show notifications
- File Manager
   - [x] List volumes
   - [x] List files
   - [x] Download file/folder
   - [x] Upload file
   - [x] Create new folder
   - [ ] Copy/move files/folders
   - [x] Delete files/folders
   - [ ] Preview image/text/sound files
- App Manager
   - [x] List installed apps
   - [x] Install APK
   - [ ] Install APKs, apkm, xapk
   - [x] Uninstall app
   - [x] Launch app on device
   - [x] Backup/restore app data
- Scrcpy
   - [x] Mirror display
   - [x] Forward sound
   - [x] Control device with mouse, pen and touchscreen
   - [x] Control device with keyboard
   - [ ] AoA mode will be deprecated, HID emulation (https://github.com/Genymobile/scrcpy/issues/4034) will be added
- Advanced
   - [x] Clear and dump logcat
   - [x] Dump bugreport
   - [x] Power menu
- Other
   - [x] Offline support
   - [ ] Desktop client with Tauri    

## Try it now

The new app is currently hosted at https://tangoapp.dev/

## Contributions

To separate issues between Tango library and the new Tango app, you can post feature requests and bug reports for the new app to the new repository:

https://github.com/tango-adb/feedback/issues

This repository's issue tracker will be reserved only for the library itself. In both cases, make sure to select the right template for your issue and provide as much information as you can.

## Future of this repository

The core library will continue to be updated. Feature requests and bug reports will be handled, and some features in the new app will be upsteamed into the core library.

The existing demo will continue to be available, but **not** updated. The plan is to create a new demo based on Storyboard.js (https://storybook.js.org/) to better showcase each feature.

The domain name https://tangoapp.dev has been switched to the new app.

The repository will be moved to the new organization at some point. This will cause the old url https://yume-chan.github.io/ya-webadb/ to stop working.

## Finalize

Again, thanks to all people contributing to this project by posting bug reports and feature requests, sending code fixes, spreading it online, and donating!

**Comment by upintheairsheep** [comment]
Make a way to bypass the first prompt for users wanting to test out without a device

**Comment by upintheairsheep** [comment]
Can you rename the legacy Tango to “Tango Legacy”?
